lib/SILOptimizer/Transforms/TempRValueElimination.cpp

```diff
@@ -419,8 +419,13 @@ bool TempRValueOptPass::extendAccessScopes(
         return false;
     }
   }
-  if (endAccessToMove)
+  if (endAccessToMove) {
+    // Don't move instructions beyond the block's terminator.
+    if (isa<TermInst>(lastUseInst))
```

Comment:
We should be able to early exit at : https://github.com/apple/swift/blob/3d950362b42650534377013f5f310647b709d6a2/lib/SILOptimizer/Transforms/TempRValueElimination.cpp#L406